src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/ManageController.cs

```diff
@@ -272,6 +281,49 @@ public async Task<IActionResult> SetPassword(SetPasswordViewModel model)
             return this.RedirectToAction(nameof(this.SetPassword));
         }
 
+        public IActionResult AuthorizeAccessToGoogleCalendar()
+        {
+            return this.Challenge(
+                new AuthenticationProperties
+                {
+                    RedirectUri = this.Url.Action(nameof(this.AuthorizationCallback), "Manage")
+                },
+                "Calendar");
+        }
+
+        public async Task<IActionResult> AuthorizationCallback()
+        {
+            var response = await this._googleCalendarService.CreateNewCalendarWithEvents();
+            this.StatusMessage = MapStatusToMessage(response.Status);
+            return this.RedirectToAction(nameof(this.GoogleCalendarIntegration));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GoogleCalendarIntegration()
+        {
+            var userHasPermissions = await this.CheckIfUserHasPermissions();
+            var model = new GoogleCalendarViewModel
+            {
+                HasPermissions = userHasPermissions,
+                StatusMessage = this.StatusMessage
```

Comment:
what should be the value of _this.StatusMessage_?

Reply:
If we're first integrating the value will be
![image](https://user-images.githubusercontent.com/28907927/95432708-4df68500-094f-11eb-8e16-46f8e58f4bcb.png)
If we've been successfully integrated the value of this.StatusMessage will be null, so nothing will be shown to the user